### ISSUE TITLE
Service failed with 'Error while checking authentication token' in cobbler log

### ISSUE THREAD

**Issue by kibardin-s** [uyuni-project/uyuni#5893]
# Problem description

Service failed with 'Error while checking authentication token' in cobbler log

## Version of Uyuni Server

Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : uyuni-server-stable
Name           : Uyuni-Server-release
Version        : 2022.08-185.2.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : Uyuni-Server-release-2022.08-185.2.uyuni2.src
Summary        : Uyuni Server

## Details about the issue
Service was unavailable with 503 http-error, in /var/log/cobbler/cobbler.log was:
2022-09-02T15:23:19 - ERROR | Error while checking authentication token.
2022-09-02T15:23:19 - ERROR | Error while checking authentication token.
2022-09-02T15:23:19 - INFO | Exception occured: <class 'xmlrpc.client.ProtocolError'>
2022-09-02T15:23:19 - INFO | Exception occured: <class 'xmlrpc.client.ProtocolError'>
2022-09-02T15:23:19 - INFO | Exception value: <ProtocolError for aaa.bbb.ccc.local/rpc/api: 500 Internal Server Error>
2022-09-02T15:23:19 - INFO | Exception value: <ProtocolError for aaa.bbb.ccc.local/rpc/api: 500 Internal Server Error>
2022-09-02T15:23:20 - INFO | Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/modules/authentication/spacewalk.py", line 68, in __check_auth_token
    return xmlrpc_client.auth.checkAuthToken(username, password)

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1112, in __call__
    return self.__send(self.__name, args)

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1452, in __request
    verbose=self.__verbose

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1154, in request
    return self.single_request(host, handler, request_body, verbose)

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1187, in single_request
    dict(resp.getheaders())

2022-09-02T15:23:20 - INFO | Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/modules/authentication/spacewalk.py", line 68, in __check_auth_token
    return xmlrpc_client.auth.checkAuthToken(username, password)

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1112, in __call__
    return self.__send(self.__name, args)

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1452, in __request
    verbose=self.__verbose

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1154, in request
    return self.single_request(host, handler, request_body, verbose)

  File "/usr/lib64/python3.6/xmlrpc/client.py", line 1187, in single_request
    dict(resp.getheaders())

**Comment by mackdk** [comment]
Have you tried to restart the spacewalk services? There is a service updating secrets in `rhn.conf`. At first glance, it looks like it could be the root cause.

**Comment by kibardin-s** [comment]
Yes I did. After restart it works, but after new 31 systems connected service failed again. It failed after 31 and after 62 connected system.

**Comment by mackdk** [comment]
> Restart spacewalk-wait-for-salt.service spacewalk-wait-for-jabberd.service spacewalk-wait-for-tomcat.service spacewalk-wait-for-taskomatic.service did not help.
Seems like tomcat.service restart make uyuni up state.

You can do `spacewalk-service restart` to restart all the services required by uyuni at once.

> after new 31 systems connected service failed again. It failed after 31 and after 62 connected system.

Not sure to have understood correctly: you see the same `Error while checking authentication token` every time you connect 31 new systems?

**Comment by mcalmer** [comment]
Closing outdated issue
